Vagrantfile

```diff
@@ -1,26 +1,205 @@
     SHELL
   end
+
+  config.vm.define "rootful" do |m|
+    m.vm.box = "bento/fedora-42"
+    m.vm.synced_folder '.', '/vagrant', disabled: true
+
+    m.vm.provider "virtualbox" do |v|
+      v.memory = 2048
+      v.cpus = 2
+    end
+    m.vm.provision "shell", path: "./hack/set_root_login_for_vagrant.sh"
+    m.vm.provision "shell", inline: <<-SHELL
+      set -e -u -o pipefail
+      yum update -y
+      yum install -y git gcc docker wget pkg-config systemd-devel dbus-devel elfutils-libelf-devel libseccomp-devel clang-devel openssl-devel just
+      grubby --update-kernel=ALL --args="systemd.unified_cgroup_hierarchy=0"
+      service docker start
+      curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
+      echo "export PATH=$PATH:$HOME/.cargo/bin" >> ~/.bashrc
+    SHELL
+    m.ssh.username = 'root'
+    m.ssh.insert_key = 'true'
+  end
+
+  config.vm.define "containerd2youki" do |m|
+    m.vm.box = "bento/ubuntu-24.04"
+    m.vm.synced_folder '.', '/vagrant/youki', disabled: false
+
+    m.vm.provider "virtualbox" do |v|
+      v.memory = 4096
+      v.cpus = 4
+    end
+
+    m.vm.provision "bootstrap", type: "shell" do |s|
+      s.inline = <<-SHELL
+        set -e -u -o pipefail
+        export DEBIAN_FRONTEND=noninteractive
+        apt-get update && apt-get install -y \
+          make \
+          pkg-config         \
+          libsystemd-dev     \
+          libdbus-glib-1-dev \
+          build-essential    \
+          libelf-dev \
+          libseccomp-dev \
+          libbtrfs-dev \
+          btrfs-progs
+        
+        ARCH=$(uname -m)
+        case "$ARCH" in
+          x86_64)
+            GOARCH="amd64"
+            ;;
+          aarch64)
+            GOARCH="arm64"
+            ;;
+          *)
+            echo "Unsupported architecture: $ARCH"
+            exit 1
+            ;;
+        esac
+        
+        wget --quiet https://go.dev/dl/go#{CONTAINERD_2_YOUKI_GO_VERSION}.linux-${GOARCH}.tar.gz -O /tmp/go#{CONTAINERD_2_YOUKI_GO_VERSION}.linux-${GOARCH}.tar.gz
+        rm -rf /usr/local/go && tar -C /usr/local -xzf /tmp/go#{CONTAINERD_2_YOUKI_GO_VERSION}.linux-${GOARCH}.tar.gz
+        echo "export PATH=$PATH:/usr/local/go/bin" >> ~/.bashrc
+        echo "export GOPATH=$HOME/go" >> ~/.bashrc
+        export PATH=$PATH:$HOME/.cargo/bin:/usr/local/go/bin
+        export GOPATH=$HOME/go
+
+        git clone https://github.com/containerd/containerd \
+          /root/go/src/github.com/containerd/containerd -b v#{CONTAINERD_2_YOUKI_CONTAINERD_VERSION}
+
+        cd /root/go/src/github.com/containerd/containerd
+        make
+        make binaries
+        make install
+        ./script/setup/install-cni
+        ./script/setup/install-critools
+        rm -rf /bin/runc /sbin/runc /usr/sbin/runc /usr/bin/runc
+        ln -s /vagrant/youki/youki /usr/bin/runc
+      SHELL
+    end
+
+    m.vm.provision "test", type: "shell", run: "never" do |s|
```

Comment:
I added `run: never` to the original configuration.
This change was made because, although the justfile separates test execution and provisioning as shown below, in the original setup, the tests were also being executed during provisioning.

```
vagrant up containerd2youki
vagrant provision containerd2youki --provision-with test
```